jdaviz/core/template_mixin.py

```diff
@@ -1677,6 +1679,82 @@ def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
+        FULLY finite, uncertainties will be set to None (in the Specviz parser),
+        so this traitlet will be False in that case (and therefore no warning
+        message displayed in the plugin).
+    """
+
+    non_finite_uncertainty_mismatch = Bool(False).tag(sync=True)
+
+    # every time a data/subset selection is changed, check the data selection and
+    # its uncertainties to see if there are any finite data elements with
+    # uncertainties. Warn in plugin if this occurs.
+
+    # note! the 'n/a' traitlet doesn't exist. we are watching it here because
+    # DatasetSpectralSubsetValidMixin is also watching the exact set
+    # of real traitlets, which creates a bug where only the last inhereted method
+    # will run, so we need to spoof it like this.
+    @observe("n/a", "dataset_selected", "spectral_subset_selected")
+    def _check_non_finite_uncertainty_mismatch(self, event={}):
+
+        if not hasattr(self, 'dataset') or self.dataset_selected == '':
+            # during initial init, this can trigger before the component is initialized
+            return
+
+        if self.spectral_subset_selected == "Entire Spectrum":
+            if not hasattr(self, '_get_1d_spectrum'):
+                # only model_fitting has _get_1d_spectrum(), but this method is here
+                # instead of there  because it may eventually be used by other plugins.
+                # if that happens, move _get_1d_spectrum() somewhere more general
+                m = "_get_1d_spectrum() must be available in plugin to use"
+                " NonFiniteUncertaintyMismatchMixin"
+                raise NotImplementedError(m)
+            spec = self._get_1d_spectrum()  # this is a method in model fitting
+            flux = spec.flux
+            uncert = spec.uncertainty
+
+            if uncert is None:
+                self.non_finite_uncertainty_mismatch = False
+                return self.non_finite_uncertainty_mismatch
+        else:
+            # get selected subset
+            spec = self._apply_subset_masks(self._get_1d_spectrum(), self.spectral_subset)
+            flux = spec.flux[~spec.mask]
+            uncert = spec.uncertainty
+            if uncert is None:
+                self.non_finite_uncertainty_mismatch = False
+                return self.non_finite_uncertainty_mismatch
+            uncert = spec.uncertainty[~spec.mask]
+
+        uncert = uncert.array
+
+        if not np.any(uncert):
+            self.non_finite_uncertainty_mismatch = False
+            return self.non_finite_uncertainty_mismatch
+
+        flux = flux.value
+
+        for i, val in enumerate(uncert):
+            if not np.isfinite(uncert[i]) and np.isfinite(flux[i]):
+                self.non_finite_uncertainty_mismatch = True
+                break  # we only need to catch this once then can stop looking
+            else:
+                self.non_finite_uncertainty_mismatch = False
```

Comment:
This should only happen if the for-loop ends without entering the if-statement and breaking (if its _in_ the for-loop, this will result in excessive calls to update the traitlet and possible UI-flickering)

```suggestion
        else:
            self.non_finite_uncertainty_mismatch = False
```

Reply:
ok, does my most recent change fix this? i think it should

Reply:
with the return instead of the break?  That should have the same result, yes